### PR TITLE
Isaiaih: Implement Loading State On Button Component

### PR DESCRIPTION
**Pull request by Mensaiah**
#### What does this PR achieve?
- modifies some CSS to perfect button component
- performs rebase  against `develop` successfully
- extracts button styles from SignUp form to new Button atom
- adds the new Button atom to SignUp form
- uses a Loader component to create loading state for Button atom

#### Any background context?
- Extracting the Button component makes re-usability possible 

#### Manual Testing Step
- When you try to Signup a user you should see a loader as shown in screenshots

#### Screenshots:
Screenshot 2019-12-27 at 12 40 46 PM
Screenshot 2019-12-27 at 12 40 38 PM
Screenshot 2019-12-27 at 12 49 41 PM

-28a6-11ea-8112-d960711e89be.png">

#### Relevant Trello Tickets
- [Button Component](https://trello.com/c/ChLtFlKJ/39-implement-loading-states-on-button-component)

### CODE REVIEW

**Comment by chukwuemekachm** [comment]
Hey, @Mensaiah can you take a look at this?
The spinner makes the button bigger. Can we retain the button size irrespective of whether we have a spinner or a text?

Screenshot 2019-11-28 at 2 20 54 PM

File: src/components/ui/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      outline: none;
+    }
+
+    &:hover {
+      cursor: pointer;
+      background: ${BRAND_PRIMARY_HOVER};
+    }
+  }`
+
+  const mapStateToProps = state => {
```
Comment (chukwuemekachm) [inline src/components/ui/Button/index.tsx]: Why is the button subscribed to the store here, what happens when the use this button outside an auth component?

File: src/components/ui/Loader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+.loader {
+  border: 3px solid ${BRAND_PRIMARY};  
+  border-top: 3px solid ${BRAND_WHITE};
+  border-radius: 50%;
+  width: ${(props: LoaderWrapperProps) => props.size === 'small' ? '25px' : '150px'};
+  height: ${(props: LoaderWrapperProps) => props.size === 'small' ? '25px' : '150px'};
+  animation: spin 2s linear infinite;
+}
+
+@keyframes spin {
```
Comment (chukwuemekachm) [inline src/components/ui/Loader/index.tsx:36]: Nice. 👍🏽

File: src/store/configure_store.ts [hunk 3]
```diff
@@ -7,6 +7,12 @@ import createRootReducer from './reducers';
 import { logger, crashReporter } from './custom_middlewares';
 import rootSaga from './sagas';
 
+declare global {
```
Comment (chukwuemekachm) [inline src/store/configure_store.ts:10]: Nice. 👍🏽

**Review by chukwuemekachm — COMMENTED** [review]
Nice one.

**Comment by Mensaiah** [comment]
Update has been made

File: src/components/ui/Button/index.tsx (hunk 1 above)
Comment (chidioguejiofor) [reply src/components/ui/Button/index.tsx]: Make sense. Fixed

**Review by chukwuemekachm — APPROVED** [review]

**Pull request merged**